cmake/onnxruntime_csharp.cmake

```diff
@@ -18,22 +18,34 @@ if (onnxruntime_USE_DNNL)
   STRING(APPEND CSHARP_PREPROCESSOR_DEFINES "USE_DNNL,")
 endif()
 
-if (onnxruntime_USE_TENSORRT)
-  STRING(APPEND CSHARP_PREPROCESSOR_DEFINES "USE_TENSORRT,")
+if (onnxruntime_USE_DML)
+  STRING(APPEND CSHARP_PREPROCESSOR_DEFINES "USE_DML,")
 endif()
 
```

Comment:
I am not sure how any of the EP specific code in the C# library that relies on these defines was tested.